### PR TITLE
Refactor the LangtorchHub to accept a module registry 

### PR DESCRIPTION
**Pull request by rubio-He**
Refactor the LangtorchHub to accept a module registry  and fix the typo in ConversationMemory.java. #138 

# TODO
- [ ] Passing in the args into registry and process them

### CODE REVIEW

File: src/main/java/ai/knowly/langtorch/hub/LangtorchHub.java [hunk 1]
```diff
@@ -1,43 +1,15 @@
...
-    if (openAIKeyConfig.isReadFromEnvFile()) {
-      return new OpenAIServiceConfigWithImplicitAPIKeyModule();
-    }
-    if (!openAIKeyConfig.getOpenAiApiKey().isPresent()) {
-      throw new IllegalArgumentException(
-          "OpenAI API key is not present. Please provide the API key in the config.");
-    }
-
-    return new OpenAIServiceConfigWithExplicitAPIKeyModule(openAIKeyConfig.getOpenAiApiKey().get());
+  public LangtorchHub(LangtorchHubModuleRegistry registry) {
```
Comment (li2109) [inline src/main/java/ai/knowly/langtorch/hub/LangtorchHub.java:11]: add `@Inject`?

File: src/main/java/ai/knowly/langtorch/example/SimpleChatBotWithExplicitKey.java [hunk 2]
```diff
@@ -4,44 +4,21 @@
...
-                  }
-
-                  @Provides
-                  ConversationMemory provideConversationMemory() {
-                    return ConversationMemory.geDefaultInstance();
-                  }
-                }));
+    LangtorchHubModuleRegistry registry = LangtorchHubModuleRegistry.create();
+    registry.registerOpenAiModule(
+        OpenAIKeyConfig.builder().setReadFromEnvFile(false).setOpenAiApiKey(openAIKey).build());
```
Comment (li2109) [inline src/main/java/ai/knowly/langtorch/example/SimpleChatBotWithExplicitKey.java:20]: I feel it's bit verbose that user needs to set two mutually exclusive fields.
I am not sure which one is considered better option:
1. Explicit key always has higher priority
2.  do validation inside build() method: if ReadFromEnvFile and OpenAiApiKey are both set, thrown an exception?

**Review by li2109 — APPROVED** [review]

File: src/main/java/ai/knowly/langtorch/hub/LangtorchHub.java (hunk 1 above)
Comment (rubio-He) [reply src/main/java/ai/knowly/langtorch/hub/LangtorchHub.java:14]: Done!

File: src/main/java/ai/knowly/langtorch/example/SimpleChatBotWithExplicitKey.java (hunk 2 above)
Comment (rubio-He) [reply src/main/java/ai/knowly/langtorch/example/SimpleChatBotWithExplicitKey.java]: thanks, agree with the point that pattern is verbose. Actually, I want one more step further. I'd like to disable the builder/setter to disallow user set both readFromEnv/ApiKey but only create specific type config, with the benefit of those factory methods we can skip option-check in the following functions.

**Pull request merged**